src/tools/compiletest/src/autofix.rs

```diff
@@ -0,0 +1,70 @@
+use rustfix::{get_suggestions_from_json, Replacement};
+use std::collections::HashSet;
+use std::error::Error;
+
+pub fn run_rustfix(code: &str, json: &str) -> String {
+    let suggestions = get_suggestions_from_json(&json, &HashSet::new())
+        .expect("could not load suggestions");
+
+    let mut fixed = code.to_string();
+
+    for sug in suggestions.into_iter().rev() {
+        for sol in sug.solutions {
+            for r in sol.replacements {
+                fixed = apply_suggestion(&mut fixed, &r)
+                    .expect("could not apply suggestion");
+            }
+        }
+    }
+
+    fixed
+}
+
+fn apply_suggestion(
+    file_content: &mut String,
+    suggestion: &Replacement,
+) -> Result<String, Box<Error>> {
+    use std::cmp::max;
+
+    let mut new_content = String::new();
+
+    // Add the lines before the section we want to replace
+    new_content.push_str(&file_content
```

Comment:
I feel like this API should be within rustfix itself, ideally shared with the rest of the code.

This also allocates once per suggestion.

Perhaps rustfix should have an API where it applies all suggestions in order, with a callback for asking for confirmation and stuff (which we pass as an empty callback here)


The purpose of this is both to test rustfix and the suggestions, so we need to use the same code as rustfix here.

Reply:
Yes, absolutely. This is just the simplest code that works.